src/api/pc_api.go

```diff
@@ -208,6 +208,210 @@ func (api *PcApi) StopProcesses(c *gin.Context) {
+	for _, state := range states.States {
+		if state.Namespace == name {
+			names = append(names, state.Name)
+		}
+	}
+	if len(names) == 0 {
+		c.JSON(http.StatusNotFound, gin.H{"error": "no processes in namespace: " + name})
+		return
+	}
+
+	results := make(map[string]string)
+	failures := 0
+	for _, processName := range names {
+		cfg, err := api.project.GetProcessInfo(processName)
+		if err != nil {
+			results[processName] = err.Error()
+			failures++
+			continue
+		}
+		cfg.Disabled = true
+		if err := api.project.UpdateProcess(cfg); err != nil {
+			results[processName] = err.Error()
+			failures++
+		} else {
+			results[processName] = "ok"
+		}
+	}
+	if failures > 0 {
+		c.JSON(http.StatusBadRequest, results)
+		return
+	}
+	c.JSON(http.StatusOK, results)
+}
+
+// @Schemes
+// @Id				EnableNamespace
+// @Description	Enables all processes in the given namespace
+// @Tags			Namespace
+// @Summary		Enable all processes in a namespace
+// @Produce		json
+// @Param			name	path		string				true	"Namespace Name"
+// @Success		200		{object}	map[string]string	"All processes in namespace enabled"
+// @Success		400		{object}	map[string]string	"Some processes in namespace failed to be enabled"
+// @Failure		404		{object}	map[string]string	"No processes in namespace"
+// @Router			/namespace/enable/{name} [patch]
+func (api *PcApi) EnableNamespace(c *gin.Context) {
+	ns := c.Param("name")
+	states, err := api.project.GetProcessesState()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	names := make([]string, 0)
+	for _, st := range states.States {
+		if st.Namespace == ns {
+			names = append(names, st.Name)
+		}
+	}
+	if len(names) == 0 {
+		c.JSON(http.StatusNotFound, gin.H{"error": "no processes in namespace: " + ns})
+		return
+	}
+
+	results := make(map[string]string)
+	failures := 0
+	for _, name := range names {
+		cfg, err := api.project.GetProcessInfo(name)
+		if err != nil {
+			results[name] = err.Error()
+			failures++
+			continue
+		}
+		cfg.Disabled = false
+		if err := api.project.UpdateProcess(cfg); err != nil {
+			results[name] = err.Error()
+			failures++
+		} else {
+			results[name] = "ok"
+		}
+	}
+	if failures > 0 {
+		c.JSON(http.StatusBadRequest, results)
+		return
+	}
+	c.JSON(http.StatusOK, results)
+}
+
+// @Schemes
+// @Id				UpdateNamespace
+// @Description	Merge processes from a partial config; all must share the same namespace/
+// @Tags			Namespace
+// @Summary		Post config fragment with processes (single namespace)
+// @Accept		json
+// @Produce		json
+// @Param			processes	body	types.Processes	true	"One or more processes, all in the same namespace"
+// @Success		200		{object}	map[string]string	"All updated"
+// @Failure		400		{object}	map[string]string  "Some processes failed to be updated. Returns error if all failed, else returns success and failures map"
+// @Router			/namespace [post]
+func (api *PcApi) UpdateNamespace(c *gin.Context) {
```

Comment:
This looks redundant - the namespace is part of the process - this should handle **all** the requested processes. It's on the client to make sure that all the processes belong to the same NS.

Reply:
Ok. Sound good.